src/Sylius/Behat/Context/Setup/PaymentContext.php

```diff
@@ -49,37 +50,39 @@
     private $paymentMethodTranslationFactory;
 
     /**
-     * @var PaymentMethodNameToGatewayConverterInterface
+     * @var ObjectManager
      */
-    private $paymentMethodNameToGatewayConverter;
+    private $objectManager;
 
     /**
-     * @var ObjectManager
+     * @var array
      */
-    private $objectManager;
+    private $gatewayFactories;
 
     /**
      * @param SharedStorageInterface $sharedStorage
      * @param PaymentMethodRepositoryInterface $paymentMethodRepository
-     * @param FactoryInterface $paymentMethodFactory
+     * @param PaymentMethodFactoryInterface $paymentMethodFactory
      * @param FactoryInterface $paymentMethodTranslationFactory
-     * @param PaymentMethodNameToGatewayConverterInterface $paymentMethodNameToGatewayConverter
      * @param ObjectManager $objectManager
      */
     public function __construct(
         SharedStorageInterface $sharedStorage,
         PaymentMethodRepositoryInterface $paymentMethodRepository,
-        FactoryInterface $paymentMethodFactory,
+        PaymentMethodFactoryInterface $paymentMethodFactory,
         FactoryInterface $paymentMethodTranslationFactory,
-        PaymentMethodNameToGatewayConverterInterface $paymentMethodNameToGatewayConverter,
         ObjectManager $objectManager
     ) {
         $this->sharedStorage = $sharedStorage;
         $this->paymentMethodRepository = $paymentMethodRepository;
         $this->paymentMethodFactory = $paymentMethodFactory;
         $this->paymentMethodTranslationFactory = $paymentMethodTranslationFactory;
-        $this->paymentMethodNameToGatewayConverter = $paymentMethodNameToGatewayConverter;
         $this->objectManager = $objectManager;
+        $this->gatewayFactories = [
```

Comment:
Can't we inject this array? If we would like to use this setup context in some project with different payment method we will have to copy whole class.